server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/issue/ShortBranchIssuesLoader.java

```diff
@@ -54,4 +61,38 @@ public ResolvedShortBranchIssuesLoader(ShortBranchComponentsWithIssues shortBran
         .collect(Collectors.toList());
     }
   }
+
+  public Map<ShortBranchIssue, DefaultIssue> loadDefaultIssuesWithChanges(Collection<ShortBranchIssue> lightIssues) {
+    if (lightIssues.isEmpty()) {
+      return Collections.emptyMap();
+    }
+    Map<String, ShortBranchIssue> issuesByKey = lightIssues.stream().collect(Collectors.toMap(ShortBranchIssue::getKey, i -> i));
+    try (DbSession session = dbClient.openSession(false)) {
+
+      Map<String, List<IssueChangeDto>> changeDtoByIssueKey = dbClient.issueChangeDao()
+        .selectByIssueKeys(session, issuesByKey.keySet()).stream().collect(groupingBy(IssueChangeDto::getIssueKey));
+
+      return dbClient.issueDao().selectByKeys(session, issuesByKey.keySet())
+        .stream()
+        .map(IssueDto::toDefaultIssue)
+        .map(i -> setChanges(changeDtoByIssueKey, i))
```

Comment:
just as curiosity, `setChanges()` wouldn't need to return the same object with `peek()`.

Reply:
Nice, I was looking for this concept, but didn't know about peek(), so ended up using map.